cmd/moby/config.go

```diff
@@ -185,39 +257,193 @@ func (m mlist) parts(i int) int {
 	return strings.Count(filepath.Clean(m[i].Destination), string(os.PathSeparator))
 }
 
-// ConfigInspectToOCI converts a config and the output of image inspect to an OCI config file
-func ConfigInspectToOCI(image *MobyImage, inspect types.ImageInspect) ([]byte, error) {
-	oci := specs.Spec{}
+// assignBool does ordered overrides from JSON bool pointers
+func assignBool(v1, v2 *bool) (ret bool) {
+	if v1 != nil {
+		ret = *v1
+	}
+	if v2 != nil {
+		ret = *v2
+	}
+	return
+}
 
-	config := inspect.Config
-	if config == nil {
-		return []byte{}, errors.New("empty image config")
+// assignBoolPtr does ordered overrides from JSON bool pointers
+func assignBoolPtr(v1, v2 *bool) (ret *bool) {
+	if v1 != nil {
+		ret = v1
+	}
+	if v2 != nil {
+		ret = v2
 	}
+	return
+}
 
-	args := append(config.Entrypoint, config.Cmd...)
-	if len(image.Command) != 0 {
-		args = image.Command
+// assignIntPtr does ordered overrides from JSON int pointers
+func assignIntPtr(v1, v2 *int) (ret *int) {
+	if v1 != nil {
+		ret = v1
+	}
+	if v2 != nil {
+		ret = v2
 	}
-	env := config.Env
-	if len(image.Env) != 0 {
-		env = image.Env
+	return
+}
+
+// assignUint32 does ordered overrides from JSON uint32 pointers
+func assignUint32(v1, v2 *uint32) (ret uint32) {
+	if v1 != nil {
```

Comment:
hmm but in the not returning a pointer case I wanted to have the default return value set by using a named value. I can still early return but I may keep the parameter